src/models/Context.js

```diff
@@ -3,15 +3,24 @@ class Context {
     this.data = data;
     this.secrets = secrets;
     this.safeData = {};
+    this.mask = mask;
+    this.maskSafeData();
+  }
+
+  maskSafeData() {
     for (const key of Object.keys(this.data)) {
       if (this.secrets[key]) {
-        this.safeData[key] = mask;
+        this.safeData[key] = this.mask;
       } else {
         this.safeData[key] = this.data[key];
       }
     }
   }
 
+  copy() {
+    return new Context({...this.data}, {...this.secrets}, this.mask);
```

Comment:
now I can't really say that this won't work, but I'm concerned about an edge case:
If data had nested JSON object, it will pass in the reference to those children, if there's a mutation down the line, it will leak. An example to demonstrate what Im trying to say:


```
let a = {x: {name:'Kyle'}}
let b = {...a}
console.log(b) // {x: {name:'Kyle'}}
a.x.name = 'khaled'
console.log(b) // {x: {name:'khaled'}}
```
you can run this snippet in node by copy pasting it (I double checked the syntax)

I suggest the library clone: 
here's a snippet you can run in [npm runkit](https://npm.runkit.com/) 
```
var clone = require('clone');
let a = {x: {name:'Kyle'}}
let b = {...a}
let c = clone(a)
console.log(b) // {x: {name:'Kyle'}}
a.x.name = 'khaled'
console.log(b) // {x: {name:'khaled'}}
console.log(c) // {x: {name:'Kyle'}}
```

Reply:
yes, good call and thanks for the suggestion about clone lib